napari/utils/transforms/transforms.py

```diff
@@ -74,6 +74,27 @@ def set_slice(self, axes: Sequence[int]) -> 'Transform':
             trans._('Cannot subset arbitrary transforms.', deferred=True)
         )
 
+    def replace_slice(
```

Comment:
I went with (1). I think it will be useful elsewhere, for example for plugins that want to set the affine property in the currently displayed dimensions.